File: javascript/58.length-of-last-word.js
/*
 * @lc app=leetcode id=58 lang=javascript
 *
 * [58] Length of Last Word
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function (s) {
  // Solution 1:
  // const strs = s.trim().split(" ");
  // return strs[strs.length - 1].length;

  // Solution 2:
  let result = 0;
  for (let i = s.length - 1; i >= 0; i--) {
    if (s[i] !== " ") result++;
    else if (result > 0) break;
  }
  return result;
};
// @lc code=end

File: rust/58.length-of-last-word.rs
/*
 * @lc app=leetcode id=58 lang=rust
 *
 * [58] Length of Last Word
 */

// @lc code=start
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        return s.split_whitespace().last().unwrap_or("").len() as i32;
    }
}
// @lc code=end

